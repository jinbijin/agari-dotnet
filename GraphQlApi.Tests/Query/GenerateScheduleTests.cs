using System.Threading.Tasks;
using GraphQlApi.Tests.Fixture;
using GraphQlApi.Tests.Variables;
using Snapshooter.Xunit;
using Xunit;

namespace GraphQlApi.Tests.Query
{
    public class GenerateScheduleTests : GraphQlApiTest<GenerateScheduleVariables>, IClassFixture<WebApiFixture>
    {
        private const string Query = "query GenerateSchedule($roundCount: Int!, $participantCount: Int!) { generateSchedule(roundCount: $roundCount, participantCount: $participantCount) { rounds { games { participantNrs } } } }";

        public GenerateScheduleTests(WebApiFixture webApiFixture) : base(webApiFixture, Query)
        {
        }

        [Fact(DisplayName = "generateSchedule OK")]
        public Task OkTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 4,
                participantCount = 20
            }, Snapshot.FullName());

        [Fact(DisplayName = "generateSchedule Error: not enough participants")]
        public Task NotEnoughParticipantsTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 4,
                participantCount = 16
            }, Snapshot.FullName());

        [Fact(DisplayName = "generateSchedule Error: not enough rounds")]
        public Task NotEnoughRoundsTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 3,
                participantCount = 20
            }, Snapshot.FullName());

        [Fact(DisplayName = "generateSchedule Error: invalid participant count")]
        public Task InvalidParticipantCountTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 4,
                participantCount = 21
            }, Snapshot.FullName());
    }
}
