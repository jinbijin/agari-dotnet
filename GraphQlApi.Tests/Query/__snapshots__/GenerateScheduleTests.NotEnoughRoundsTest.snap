{
  "ResponseHeaders": {
    "Version": "1.1",
    "Content": {
      "Headers": [
        {
          "Key": "Content-Type",
          "Value": [
            "application/json; charset=utf-8"
          ]
        },
        {
          "Key": "Content-Length",
          "Value": [
            "203"
          ]
        }
      ]
    },
    "StatusCode": "OK",
    "ReasonPhrase": "OK",
    "Headers": [],
    "TrailingHeaders": [],
    "RequestMessage": {
      "Version": "1.1",
      "VersionPolicy": "RequestVersionOrLower",
      "Content": {
        "Headers": [
          {
            "Key": "Content-Type",
            "Value": [
              "application/json; charset=utf-8"
            ]
          },
          {
            "Key": "Content-Length",
            "Value": [
              "252"
            ]
          }
        ]
      },
      "Method": {
        "Method": "POST"
      },
      "RequestUri": "http://localhost/graphql",
      "Headers": [],
      "Properties": {},
      "Options": {}
    },
    "IsSuccessStatusCode": true
  },
  "ResponseContent": {
    "data": {
      "generateSchedule": null
    },
    "errors": [
      {
        "message": "Number of rounds must be at least 4.",
        "locations": [
          {
            "line": 1,
            "column": 70
          }
        ],
        "path": [
          "generateSchedule"
        ],
        "extensions": {
          "code": "INVALID_PARAMETER"
        }
      }
    ]
  }
}
